### DISCUSSION TITLE
Underlying Cloud Infrastructure for GDPR Compliance

### DISCUSSION THREAD

**Discussion by RobinWie** [supabase/supabase#12623] [Questions]
So I was wondering whether there is some guaranteed information on which cloud infrastructure provider is utilized under the hood when I have the hosted option (free or pro). My question is this:
As a german company, especially because we deal with very sensitive data, GDPR compliance is extremely important. Unfortunately, this makes it impossible to use Google as a cloud provider. AWS would be great, but I have to ensure that all my customer's data is hosted on AWS and AWS only, and that there are no services (e.g. edge functions or authentication) that depend on another cloud provider.
I know that e.g. the infrastructure settings in the project, as well as other discussions, suggest that AWS is being used. However, I need to guarantee that it is the only provider that's being used and that the data does not go anywhere else.
I couldn't find any official or reliable declaration on this myself, so I guess Supabase keeps this vague on purpose so that they don't have to make any promises for the future. But maybe I also missed something obvious here, so I would be happy for any insight!

ps: I know I could self-host Supabase on AWS and be safe with it, but self-hosting somewhat undermines the idea of BaaS for me, so that would be inconvenient

**Comment by devidevio** [comment]
Maybe this one helps you?
https://github.com/supabase/supabase/discussions/4815

**Reply by GaryAustin1** [reply]
Edge functions are run on Deno deploy and is not directly controlled by Supabase.  I don't know if they use AWS or not, but there is no ability to set the datacenter currently so they will normally run in a data center closest to the user.

**Reply by RobinWie** [reply]
@Unkn0wn0x already saw this one, seems like the database is entirely built on top of aws, thank you!

@GaryAustin1 That's interesting information. So basically using edge functions in Supabase might cause your data to go anywhere as far as I understand it. Have to be cautious with that, thanks !

**Reply by flowni** [reply]
@GaryAustin1 so besides edge functions, my data is only processed at the AWS location I selected at the beginning?

**Reply by DH-555** [reply]
> Edge functions are run on Deno deploy and is not directly controlled by Supabase. I don't know if they use AWS or not, but there is no ability to set the datacenter currently so they will normally run in a data center closest to the user.

I saw deno is not even GDPR complaint, it's a really big issue for apps in the EU. Since deno is open source, supabase could host an instance to make it complaint for EU users
